responses.py

```diff
@@ -332,10 +339,34 @@ def add(self, method_or_response=None, url=None, body='', *args, **kwargs):
             self._matches.append(method_or_response)
             return
 
-        if method_or_response:
-            kwargs['method'] = method_or_response
+        self._matches.append(
+            Response(method=method_or_response, url=url, body=body, **kwargs))
+
+    def remove(self, method_or_response=None, url=None):
```

Comment:
i think it'd be better to require the instance on removal, and simply create an equality check on the BaseResponse class. Alternatively it needs to remove *any* matching instance.

Reply:
I implemented the equality check, and this now removes all matching instances (it still supports passing the instance, but it's pretty convenient given the use case in the PR description to be able to pass the method and url too).

Reply:
one last bit -- could we add a basic docstring to this to show syntax and usage

same for replace below

otherwise lgtm

Reply:
@dcramer: Docstrings added, let me know what you think.